atlasdb-config/src/main/java/com/palantir/atlasdb/config/LeaderConfig.java

```diff
@@ -65,8 +74,10 @@ public long leaderPingResponseWaitMs() {
 
     @Value.Check
     protected final void check() {
-        Preconditions.checkArgument(leaders().contains(localServer()),
-                "The localServer '%s' must included in the leader entries %s.", localServer(), leaders());
+        Preconditions.checkArgument(!isLeader() || localServer().isPresent(),
+                "Local server isn't set even though we are set to be a leader");
+        Preconditions.checkArgument(leaders().size() == leaderCount(),
+                "Number of leaders {} does not equal the expected leader count {}", leaders().size(), leaderCount());
```

Comment:
Followed up offline with Bobby about how I want this for catching misconfigurations in leader lists.